src/main/kotlin/com/ampnet/identityservice/service/impl/ReCaptchaServiceImpl.kt

```diff
@@ -0,0 +1,89 @@
+package com.ampnet.identityservice.service.impl
+
+import com.ampnet.identityservice.config.ApplicationProperties
+import com.ampnet.identityservice.exception.ReCaptchaException
+import com.ampnet.identityservice.service.ReCaptchaService
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+import org.springframework.web.util.UriComponentsBuilder
+import java.net.URI
+
+@Service
+class ReCaptchaServiceImpl(
+    private val applicationProperties: ApplicationProperties,
+    private val restTemplate: RestTemplate,
+    private val objectMapper: ObjectMapper
+) : ReCaptchaService {
+
+    @Throws(ReCaptchaException::class)
+    override fun validateResponseToken(reCaptchaToken: String?) {
+        if (applicationProperties.reCaptcha.enabled.not()) return
+        if (reCaptchaToken.isNullOrEmpty())
+            throw ReCaptchaException("reCAPTCHA verification is enabled, send a valid token")
+        try {
+            val responseEntity = restTemplate.postForEntity<String>(generateGoogleUri(reCaptchaToken))
+            val googleResponse = readGoogleResponse(responseEntity)
+            if (responseEntity.statusCode.is2xxSuccessful) {
+                if (googleResponse.success) {
```

Comment:
You can un-nest this if: `if (responseEntity.statusCode.is2xxSuccessful && googleResponse.success)`